### PR TITLE
feat: add new model `TourStop` and its GraphQL

### PR DESCRIPTION
**Pull request by donni106**
- refactored duplicated attributes and functions in subclasses to their main `attraction` class
- created new subclass `TourStop` of `attraction` that belongs to `Tour`
- created new table to connect tours with multiple tour stops
- added a `payload` store to attractions to be able to save more custom data like we implemented already for generic items
- updated credentials because of a missing value for `default from:` in `ApplicationMailer`
- auto annotated models with missing fields
- created new types for new model
- added new argument for tour stops in mutation for creating tours
- added the new `payload` field in all attraction types

refactor: tour_tour_stops to tour_stop_assignments

- changed naming for clarity and readability
- added missing spec for relations of tour_stop_assignments

SVA-572

---

when querying for a tour with tour stops:

Bildschirmfoto 2022-07-26 um 19 46 03

### CODE REVIEW

File: app/models/data_resources/tour_stop.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+#
+# All stops of a tour which are interesting and have more detailed data
+#
+class TourStop < Attraction
+  has_one :tour_tour_stop, class_name: "TourTourStop", foreign_key: :tour_stop_id
+  has_one :tour, through: :tour_tour_stop
```
Comment (donni106) [inline app/models/data_resources/tour_stop.rb:8]: i am not sure if this is the best way, but it works. any better suggestions?

File: app/models/data_resources/attraction.rb [hunk 2]
```diff
@@ -4,11 +4,16 @@
 # Attraction is the superclass for all (touristic) attractions in the smart village.
 #
 class Attraction < ApplicationRecord
+  include FilterByRole
+
+  attr_accessor :force_create
```
Comment (marcometz) [inline app/models/data_resources/attraction.rb:9]: 👍

**Comment by marcometz** [comment]
Das Refactoring aus den Tour und PointOfInterest Models raus zu Attractions rein hätte ich lieber in einem separaten Commit gehabt zum reviewen

File: app/models/data_resources/tour.rb [hunk 3]
```diff
@@ -5,46 +5,27 @@
...
-
-  attr_accessor :force_create
   enum means_of_transportation: { bike: 0, canoe: 1, foot: 2 }
 
-  belongs_to :data_provider
-
   has_many :data_resource_categories, -> { where(data_resource_type: "Tour") }, foreign_key: :data_resource_id
   has_many :categories, through: :data_resource_categories
   has_many :geometry_tour_data, as: :geo_locateable, class_name: "GeoLocation", dependent: :destroy
+  has_many :tour_tour_stops, class_name: "TourTourStop", foreign_key: :tour_id, dependent: :destroy
```
Comment (marcometz) [inline app/models/data_resources/tour.rb:13]: die Kreutzabelle müsste eigentlich tours_tour_stops heißen. Durch die Doppelung

```
has_and_belongs_to_many associative tables generally have the naming convention of [plural_first_alphabetical_model]_[plural_second_alphabetical_model] with any shared naming prepended
```

Finde ich durch die Doppelung mit Tour aber irgendwie auch nicht viel besser.
Was hältst du von `tour_stop_assignments`und demzufolge das Model auch TourStopAssignment

File: app/models/data_resources/tour_stop.rb (hunk 1 above)
Comment (marcometz) [inline app/models/data_resources/tour_stop.rb:8]: und dann hier sowas wie:

```
has_one :tour_stop_assignment, class_name: "TourStopAssignment", foreign_key: :tour_stop_id
has_one :tour, through: :tour_stop_assignment
```

File: app/models/tour_tour_stop.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class TourTourStop < ApplicationRecord
```
Comment (marcometz) [inline app/models/tour_tour_stop.rb:3]: siehe zuvor: TourStopAssignment ?

Comment (donni106) [reply app/models/tour_tour_stop.rb]: done with bfe6031

File: app/models/data_resources/tour_stop.rb (hunk 1 above)
Comment (donni106) [reply app/models/data_resources/tour_stop.rb]: done with bfe6031

File: app/models/data_resources/tour.rb (hunk 3 above)
Comment (donni106) [reply app/models/data_resources/tour.rb]: I like the idea, done with bfe6031

File: app/models/data_resources/tour_stop.rb (hunk 1 above)
Comment (donni106) [reply app/models/data_resources/tour_stop.rb]: refactored with bfe6031

**Review by marcometz — APPROVED** [review]

**Review by marcometz — APPROVED** [review]

**Pull request merged**